src/proxy/server.rs

```diff
@@ -62,17 +68,25 @@ struct RunRecvFromArgs {
     socket: Arc<UdpSocket>,
     session_manager: SessionManager,
     session_ttl: Duration,
-    send_packets: mpsc::Sender<Packet>,
+    send_packets: mpsc::Sender<UpstreamPacket>,
     shutdown_rx: watch::Receiver<()>,
 }
 
+/// Packet received from local port
+#[derive(Debug)]
+struct DownstreamPacket {
+    from: EndpointAddress,
```

Comment:
```suggestion
    from: EndpointAddress,
```
nit: let's call this `source` rather than `from` - since we already talk about src/dst instead of from/to?

Reply:
my preference would be to use src/dest and we can go through the code base and update usages. but in any case neither change is related to this PR and we can leave this as is

Reply:
I actually don't have strong opinions. src/dest is fine by me.

Reply:
👍 on source and destination.